### ISSUE TITLE
Support running a chain of services

### ISSUE THREAD

**Issue by qiuwei** [EUDAT-GEF/GEF#11]
Data retrieval and storage can be then implemented as one service in the chain.
This also provides great flexibility for other usecases.

The chains can use named docker volumes as the media for data exchange.

**Comment by qiuwei** [comment]
A chain is defined as
a ordered sequence of jobs.
A job is defined as a container has two volumes bound, one for input, the other for output.

**Comment by qiuwei** [comment]
What is a workflow?
A graph consists of jobs?(do we have any restrictions on the structure of the graph? linear?)

What is a job?
a Job is a container with volumes bound. Only one volume is allowed for output.(Should it be 1-to-1 map? probably yes)
But should we allow multiple volumes for input?(probably yes)
Are volumes allowed to bind to arbitrary paths inside the container? (probably yes)
Are input volumes read-only? (yes!)
Should we allow reading and writing to a single volume at the same time? (probably not)

**Comment by emanueldima** [comment]
I agree that it makes sense to do this, but we also must implement something fast. So supporting a chain should not be a priority for immediate development (until February'17).

**Comment by qiuwei** [comment]
Yes, you are right.

But workflow is essential to make data retrieval work.
Anyway, a workflow is just an ordered sequence of jobs.
